### ISSUE TITLE
Calculate B1map and MPM maps without SPM environment

### ISSUE THREAD

**Issue by franciscofritz** [hMRI-group/hMRI-toolbox#106]
I would like to ask if it is a way to calculate the B1map (using the EPI method) and the MPM maps without using neither the SPM batch or SPM batch-derived script. Or, in other words, if there are specific .m functions from the hMRI toolbox that can be called and/or modified in a different script, either in Matlab or Python, without using the SPM job file approach.

We are currently analysing quite a big dataset which has been standarised using BIDS-conform approach (with bidscoin software). However, we are finding a lot of problems with the nifti outpus, either the problem comes from the metadata (which it is sorted out in the last release) or the hard-coded B1map calculation (e.g. expecting the data to be loaded as a pair SE-STE per angle or being single 3D volumes instead of two 4D volumes with SE and STE data, the mixing time not defined as the difference between the two echo times, etc). And we need to iterate for several subjects, sessions and runs, which makes these problems difficult to deal one by one.

**Comment by lukeje** [comment]
No, the B1 mapping is all written within the SPM job manager environment: https://github.com/hMRI-group/hMRI-toolbox/blob/master/hmri_create_b1map.m

However all your problems sound like they are due to malformed BIDS. If you get your BIDS sidecar right (and follow the hints [here](https://github.com/hMRI-group/hMRI-toolbox/wiki/BIDS)), everything should work out of the box.

**Comment by lukeje** [comment]
@Barisevrenugur That doesn't work for qMRI BIDS 3D EPI SE/STE, as the echo times and flip angles must be provided in the sidecar file for each individual volume.

**Comment by Barisevrenugur** [comment]
closed as answered
